frontend/src/hooks/useTranslateQuality.ts

```diff
@@ -0,0 +1,24 @@
+import { Quality } from '@/bindings/definitions';
+import { useTranslation } from 'react-i18next';
+
+/**
+ * Returns a function that translates values of the Quality enum.
+ */
+export function useTranslateQuality() {
+  const { t } = useTranslation('enums', { keyPrefix: 'Quality' });
+
+  return (key: Quality) => {
+    // We use a switch case here to avoid having to fight with
+    // the type system.
+    switch (key) {
+      case 'organic':
+        return t('Organic');
+      case 'not organic':
+        return t('NotOrganic');
+      case 'unknown':
+        return t('Unknown');
+      default:
+        throw 'Unknown enum key';
```

Comment:
Is it maybe possible to not have this so that we get a type error if we extend the enum with another option?

Reply:
I agree its not an elegant solution but I chose this approach due to useTranslation's implementation.

All keys that are declared in a json translation document are converted to sum types by react-i18n.
This means the validity of translation keys that are passed to the t function is only checked at compile time.

All strings that are not guaranteed (in a way that is verifiable by typescripts type system) to contain only translation keys will be rejected.

Reply:
I have reimplemented this section using code that should work for all future enum members.

However, I had to turn off all compiler and linker warnings, since typescript can't automatically verify that the string given to the t function is a valid translation key.
This is also the reason why I went with the switch/case solution in the first place.

Reply:
It was only a question, not a request to change it.

I liked the previous solution better. I don't think we will often extend the enum. My question was only about if we can use the compiler to help us telling us that a change in the switch is needed.

@4ydan do the E2E tests actually tell us if JavaScript exceptions are thrown? I think the E2E tests should fail if there are unhandled exceptions.

Reply:
>My question was only about if we can use the compiler to help us telling us that a change in the switch is needed.

No, I don't think so.

Reply:
>@4ydan do the E2E tests actually tell us if JavaScript exceptions are thrown? I think the E2E tests should fail if there are unhandled exceptions.

Currently I m not looking into return statuses and thrown exceptions.

Reply:
Thx, I created #896 for that.